### PR TITLE
Bump k8s libraries to 1.21 and controller-runtime to v0.9

### PR DESCRIPTION
**Pull request by palexster**
# Description

This PR bumps the version of several libraries used by Liqo:
- [x] Kubernetes libraries to v1.21.0
- [x] Controller-runtime to v0.9.0beta1

# Tests

- [x] Unit Test
- [x] E2E Test

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <p align="center">
   <a href="https://github.com/liqotech/liqo/actions"><img src="https://github.com/liqotech/liqo/workflows/Go/badge.svg" alt="Integration Pipeline Status"></a>
-  <a href="https://goreportcard.com/github.com/liqotech/liqo"><img src=https://goreportcard.com/badge/github.com/liqotech/liqo></a>
-  <a href="https://coveralls.io/repos/github/liqotech/liqo" alt="Liqo Test Coverage"></a><img src=https://coveralls.io/repos/github/liqotech/liqo/badge.svg?branch=master></a>
-  <img src=https://app.fossa.com/api/projects/git%2Bgithub.com%2Fliqotech%2Fliqo.svg?type=shield>
+  <a href="https://goreportcard.com/report/github.com/liqotech/liqo"><img src=https://goreportcard.com/badge/github.com/liqotech/liqo></a>
+  <a href="https://coveralls.io/repos/github/liqotech/liqo" alt="Liqo Test Coverage"><img src=https://coveralls.io/repos/github/liqotech/liqo/badge.svg?branch=master></a>
```
Comment (giorio94) [inline README.md:4]: The coveralls link is not working

File: cmd/liqonet/main.go [hunk 2]
```diff
@@ -90,7 +92,7 @@ func main() {
 			klog.Errorf("unable to setup controller: %s", err)
 			os.Exit(1)
 		}
-		if err := mgr.Start(r.SetupSignalHandlerForRouteOperator()); err != nil {
+		if err := mgr.Start(ctx); err != nil {
```
Comment (giorio94) [inline cmd/liqonet/main.go:95]: AFAIU This should be reverted

File: cmd/liqonet/main.go [hunk 3]
```diff
@@ -123,7 +125,7 @@ func main() {
 			os.Exit(1)
 		}
 		klog.Info("Starting manager as Tunnel-Operator")
-		if err := mgr.Start(tc.SetupSignalHandlerForTunnelOperator()); err != nil {
+		if err := mgr.Start(ctx); err != nil {
```
Comment (giorio94) [inline cmd/liqonet/main.go:128]: Same

File: cmd/liqonet/main.go [hunk 4]
```diff
@@ -169,7 +171,7 @@ func main() {
 		go r.StartServiceWatcher()
 		go r.StartSecretWatcher()
 		klog.Info("starting manager as tunnelEndpointCreator-operator")
-		if err := mgr.Start(r.SetupSignalHandlerForTunEndCreator()); err != nil {
+		if err := mgr.Start(ctx); err != nil {
```
Comment (giorio94) [inline cmd/liqonet/main.go:174]: Same

**Review by giorio94 — APPROVED** [review]
/lgtm

Just some minor comments

**Comment by palexster** [comment]
Ref. #376

**Review by aleoli — APPROVED** [review]

**Review by alacuku — APPROVED** [review]

**Pull request merged**